nixos/modules/services/desktops/pipewire.nix

```diff
@@ -46,12 +66,127 @@ in {
         '';
       };
 
-      extraConfig = mkOption {
-        type = types.lines;
-        default = "";
+      config = mkOption {
+        type = types.attrs;
         description = ''
-          Literal string to append to /etc/pipewire/pipewire.conf.
+          Configuration for the pipewire daemon.
         '';
+        default = {
+          properties = {
+            ## configure properties in the system
+            #library.name.system = "support/libspa-support";
+            #context.data-loop.library.name.system = "support/libspa-support";
+            #link.max-buffers = 64;
+            link.max-buffers = 16;		# version < 3 clients can't handle more
+            #mem.allow-mlock = true;
+            #mem.mlock-all = false;
+            #log.level = 2;
+
+            ## Properties for the DSP configuration
+            #default.clock.rate	= 48000;
+            #default.clock.quantum = 1024;
+            #default.clock.min-quantum = 32;
+            #default.clock.max-quantum = 8192;
+            #default.video.width = 640;
+            #default.video.height =	480;
+            #default.video.rate.num = 25;
+            #default.video.rate.denom = 1;
+          };
+          pa-libs = {
+            ## <factory-name regex> = <library-name>
+            #
+            # used to find spa factory names. It maps an spa factory name
+            # regular expression to a library name that should contain
+            # that factory.
+            #
+            audio."convert*" = "audioconvert/libspa-audioconvert";
+            api.alsa."*" = "alsa/libspa-alsa";
+            api.v4l2."*" = "v4l2/libspa-v4l2";
+            api.libcamera."*" = "libcamera/libspa-libcamera";
+            api.bluez5."*" = "bluez5/libspa-bluez5";
+            api.vulkan."*" = "vulkan/libspa-vulkan";
+            api.jack."*" = "jack/libspa-jack";
+            support."*" = "support/libspa-support";
+            #videotestsrc = "videotestsrc/libspa-videotestsrc";
+            #audiotestsrc = "audiotestsrc/libspa-audiotestsrc";
+          };
+          modules = {
+            ##  <module-name> = { [args = "<key>=<value> ..."]
+            #                     [flags = [ifexists]|[nofail]}
+            #
+            # Loads a module with the given parameters.
+            # If ifexists is given, the module is ignoed when it is not found.
+            # If nofail is given, module initialization failures are ignored.
+            #
+            libpipewire-module-rtkit = {
+              args = {
+                nice.level = -11;
+                rt.prio = 20;
+                rt.time.soft = 200000;
+                rt.time.hard = 200000;
+              };
+              flags = "ifexists|nofail";
+            };
+            libpipewire-module-protocol-native = null;
```

Comment:
This needs to be ordered in order for pipewire to work on my machine (thus the sortProperties in the generation), though I'm not too sure about how nice it looks with `libpipewire-module-protocol-native = { _priority = -100; _content = "null"; };` being what I have locally.

Reply:
Is this addressed now? Do we care?